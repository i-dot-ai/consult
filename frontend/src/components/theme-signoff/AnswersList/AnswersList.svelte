<script lang="ts">
  import clsx from "clsx";

  import MaterialIcon from "../../MaterialIcon.svelte";
  import Docs from "../../svg/material/Docs.svelte";

  export interface Props {
    title: string;
    answers?: string[];
    variant?: "selected" | "generated";
  }

  let { title = "", answers = [], variant = "selected" }: Props = $props();
</script>

<div class="flex items-center gap-1">
  <MaterialIcon
    color={variant === "selected" ? "fill-primary" : "fill-emerald-700"}
    size="1.2rem"
  >
    <Docs />
  </MaterialIcon>

  <h2>{title}</h2>

  <div
    class={clsx([
      "ml-1",
      "px-1.5",
      "py-0.5",
      "rounded-lg",
      "text-xs",
      variant === "selected"
        ? "bg-pink-100 text-primary"
        : "bg-emerald-50 text-emerald-700",
    ])}
  >
    {answers.length}
  </div>
</div>

{#if answers?.length > 0}
  <ul class="mt-2 max-h-[20rem] overflow-y-auto">
    {#each answers as answer, i}
      <li
        class={clsx([
          "flex",
          "items-center",
          "gap-2",
          "mb-2",
          "last:mb-0",
          "p-2",
          "rounded-lg",
          "text-neutral-600",
          "bg-neutral-50",
        ])}
      >
        <div
          class={clsx([
            "shrink-0",
            "flex",
            "justify-center",
            "items-center",
            "w-[3ch]",
            "h-[3ch]",
            "rounded-full",
            "text-xs",
            variant === "selected"
              ? "bg-pink-200 text-primary"
              : "bg-emerald-50 text-emerald-700",
          ])}
        >
          {i + 1}
        </div>

        <p class="text-sm">
          {answer}
        </p>
      </li>
    {/each}
  </ul>
{:else}
  <p class="mt-2 text-sm text-neutral-500 italic">There are no answers</p>
{/if}
