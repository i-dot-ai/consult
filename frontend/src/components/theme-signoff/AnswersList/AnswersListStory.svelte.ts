import AnswersList from "./AnswersList.svelte";

let variant = $state("selected");
let title = $state("Test Title");
let answers = $state([
  "Answer 1",
  "Answer 2",
  "Answer 3",
]);

export default {
  name: "AnswersList",
  component: AnswersList,
  category: "Theme Signoff",
  props: [
    {
      name: "variant",
      value: variant,
      type: "select",
      options: [
        { value: "selected", label: "Selected" },
        { value: "generated", label: "Generated" },
      ],
    },
    {
      name: "title",
      value: title,
      type: "text",
    },
    {
      name: "answers",
      value: answers,
      type: "json",
    },
  ],
  stories: [
    {
      name: "No Answers",
      props: {
        variant: "selected",
        answers: [],
        title: "No Answers Story",
      },
    },
  ],
};
