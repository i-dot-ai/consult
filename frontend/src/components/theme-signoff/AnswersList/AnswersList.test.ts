import { afterEach, beforeEach, describe, expect, it } from "vitest";
import { render, cleanup } from "@testing-library/svelte";

import AnswersList, { type Props } from "./AnswersList.svelte";

let testData: Props;

describe("AnswersList", () => {
  beforeEach(() => {
    testData = {
      title: "Test Title",
      answers: ["Answer 1", "Answer 2"],
    };
  })

  afterEach(() => cleanup());

  it("should render", async () => {
    const { container, getByText, getAllByText, queryByText } = render(AnswersList, {
      ...testData,
    });

    expect(getByText("Test Title"));
    expect(getByText("Answer 1"));
    expect(getByText("Answer 2"));

    // Confirm answeer 1's number
    expect(getByText("1"));
    // Finds 2, one is total num of answers, other is answer 2's number
    expect(getAllByText("2")).toHaveLength(2);

    // Make sure number don't start at 0
    expect(queryByText("0")).toBeNull();

    expect(container).toMatchSnapshot();
  });

  it("should render not found message if no answers", async () => {
    const NOT_FOUND_MESSAGE = "There are no answers";

    const { getByText } = render(AnswersList, {
      ...testData,
      answers: undefined,
    });

    expect(getByText(NOT_FOUND_MESSAGE));
  })
});
