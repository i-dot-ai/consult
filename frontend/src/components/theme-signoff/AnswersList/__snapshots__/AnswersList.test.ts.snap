// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`AnswersList > should render 1`] = `
<div>
  <div
    class="flex items-center gap-1"
  >
    <svg
      class="fill-primary"
      height="1.2rem"
      role="img"
      viewBox="0 -960 960 960"
      width="1.2rem"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M320-440h320v-80H320v80Zm0 120h320v-80H320v80Zm0 120h200v-80H320v80ZM240-80q-33 0-56.5-23.5T160-160v-640q0-33 23.5-56.5T240-880h320l240 240v480q0 33-23.5 56.5T720-80H240Zm280-520v-200H240v640h480v-440H520ZM240-800v200-200 640-640Z"
        xmlns="http://www.w3.org/2000/svg"
      />
      <!---->
    </svg>
    <!---->
     
    <h2>
      Test Title
    </h2>
     
    <div
      class="ml-1 px-1.5 py-0.5 rounded-lg text-xs bg-pink-100 text-primary"
    >
      2
    </div>
  </div>
   
  <ul
    class="mt-2 max-h-[20rem] overflow-y-auto"
  >
    <li
      class="flex items-center gap-2 mb-2 last:mb-0 p-2 rounded-lg text-neutral-600 bg-neutral-50"
    >
      <div
        class="shrink-0 flex justify-center items-center w-[3ch] h-[3ch] rounded-full text-xs bg-pink-200 text-primary"
      >
        1
      </div>
       
      <p
        class="text-sm"
      >
        Answer 1
      </p>
    </li>
    <li
      class="flex items-center gap-2 mb-2 last:mb-0 p-2 rounded-lg text-neutral-600 bg-neutral-50"
    >
      <div
        class="shrink-0 flex justify-center items-center w-[3ch] h-[3ch] rounded-full text-xs bg-pink-200 text-primary"
      >
        2
      </div>
       
      <p
        class="text-sm"
      >
        Answer 2
      </p>
    </li>
    
  </ul>
  <!---->
  
</div>
`;
